src/AndcultureCode.CSharp.Web/Controllers/Controller.cs

```diff
@@ -0,0 +1,575 @@
+                {
+                    return headers[HttpHeaders.X_FORWARDED_FOR];
+                }
+
+                var forwardedIp = Request.GetForwardedIpAddress();
+                if (string.IsNullOrWhiteSpace(forwardedIp))
+                {
+                    return string.Empty;
+                }
+
+                return forwardedIp;
+            }
+        }
+
+        /// <summary>
+        /// Requesting user's agent
+        /// </summary>
+        /// <value></value>
+        protected string UserAgent
+        {
+            get
+            {
+                if (Request == null || Request.Headers == null || !Request.Headers.ContainsKey(HeaderNames.UserAgent))
+                {
+                    return string.Empty;
+                }
+
+                var userAgent = Request.Headers[HeaderNames.UserAgent];
+                if (string.IsNullOrWhiteSpace(userAgent))
+                {
+                    return string.Empty;
+                }
+
+                return userAgent;
+            }
+        }
+
+        #endregion Public Properties
+
+        #region Results
+
+        /// <summary>
+        /// Responds with HTTP 202 Accepted
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="errors"></param>
+        /// <typeparam name="T"></typeparam>
+        public AcceptedResult Accepted<T>(T value, IEnumerable<IError> errors)
+            => base.Accepted(CreateResult(value, errors));
+
+        /// <summary>
+        /// Responds with HTTP 400 Bad Request
+        /// </summary>
+        /// <param name="errors"></param>
+        /// <typeparam name="T"></typeparam>
+        public BadRequestObjectResult BadRequest<T>(IEnumerable<IError> errors)
+            => base.BadRequest(CreateResult<object>(null, errors));
+
+        /// <summary>
+        /// Responds with HTTP 400 Bad Request
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="errors"></param>
+        /// <typeparam name="T"></typeparam>
+        public BadRequestObjectResult BadRequest<T>(T value, IEnumerable<IError> errors)
+            => base.BadRequest(CreateResult(value, errors));
+
+        /// <summary>
+        /// Responds with HTTP 400 Bad Request
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="errors"></param>
+        /// <typeparam name="T"></typeparam>
+        public BadRequestObjectResult BadRequest<T>(T value, params IError[] errors)
+            => base.BadRequest(CreateResult(value, errors));
+
+        /// <summary>
+        /// Responds with HTTP 400 Bad Request
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="message"></param>
+        /// <param name="type"></param>
+        [ApiExplorerSettings(IgnoreApi = true)] // Tell swagger to ignore
+        public BadRequestObjectResult BadRequest(string key, string message, ErrorType type = ErrorType.Error)
```

Comment:
Weird that only this method needed the `IgnoreApi = true` attribute. Does it think that this method is an actual controller route?

Reply:
Yeah, being a public method on a controller, the swagger static analysis tools inspect it and complain that it needs to be decorated with the [Http*] Aspnet MVC decorator. I'm guessing the other "public" non-http intended methods are ignored because they use generics?

Reply:
Ahhh! That might be it. I remember running into it with a few methods on the LMS project when we did the dotnet core 2.2 upgrade, and I think my solution was to make them protected or private. Guess the generics bypassing the analysis tool makes sense.